tests/transport/test_transport_compliance.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+
+import flask
+import pytest
+from pytest_localserver.http import WSGIServer
+from six.moves import http_client
+
+
+# .invalid will never resolve, see https://tools.ietf.org/html/rfc2606
+NXDOMAIN = 'test.invalid'
+
+
+def test_server_app():
+    """A basic WSGI app that exposes methods for testing http requests."""
+    app = flask.Flask(__name__)
+
+    @app.route('/basic')
+    def index():
+        return 'Basic Content', http_client.OK, {'X-Test-Header': 'value'}
+
+    @app.route('/server_error')
+    def server_error():
+        return 'Error', http_client.INTERNAL_SERVER_ERROR
+
+    @app.route('/authorized')
+    def authorized():
+        """Echoes the request auth and user agent headers."""
+        return flask.jsonify({
+            'authorization': flask.request.headers.get('authorization'),
+            'user_agent': flask.request.headers.get('user-agent')
+        })
+
+    @app.route('/authorized_refresh')
+    def authorized_refresh():
+        """Returns UNAUTHORIZED if the request token doesn't contain
+        '-new', otherwise returns the auth and user agent headers."""
+        authorization = flask.request.headers.get('authorization', '')
+
+        if '-new' not in authorization:
+            return 'Not authorized', http_client.UNAUTHORIZED
+
+        return flask.jsonify({
+            'authorization': flask.request.headers.get('authorization'),
+            'user_agent': flask.request.headers.get('user-agent')
+        })
+
+    return app.wsgi_app
+
+
+class MockCredentials(object):
+    def __init__(self, token):
+        self.token = token
+        self.user_agent = 'test-credentials'
+
+    def _before_request(self, http, uri, headers):
+        headers[b'authorization'] = 'Bearer {}'.format(self.token)
+
+    def refresh(self, http):
+        self.token = '{}{}'.format(self.token, '-new')
+
+
+class TransportComplianceTests(object):
```

Comment:
Unfortunately no. The options are:

1. Name the class so that pytest doesn't treat it as a test
2. Name the module (test_transport_compliance) something else (transport_compliance, perhaps).